### DISCUSSION TITLE
How to use RS256 with JWT using a Public Key

### DISCUSSION THREAD

**Discussion by mudasirmirza** [mqttjs/MQTT.js#1898] [Q&A]
Hi Everyone,

I am trying to configure and use EMQX with Authentication using JWT with RS256 but I am unable to find any decent documentation on this. The only reference I found is over 4 to 5 years old.
I have a separate service that generated JWT tokens and I have a public key that I can add in my emqx.conf but unable to find a decent working example on how to use RS256 with public key using JWT.

Any assistance or any pointers will be appreciated.

**Comment by robertsLando** [comment]
Never used RS256 so I dunno what you are tring to do specifically. How are you supposed to pass the jwt to the broker? The only way I can think about is by putting the jwt in the password field

**Comment by mudasirmirza** [comment]
Hi @robertsLando ,

The JWT service generates the tokens and passes them to my IoT device and from IoT device the JWT token is used as password when connecting to EQMX. I have tested that if I generate HMAC based JWT token from my service and use that token in password field from my IoT device it works but I want to use RS256 as that is what my service is currently configured to use so changing that to HMAC will be a lot of work for me.

**Reply by robertsLando** [reply]
I don't think this is something strictly related to MQTTjs BTW

**Comment by mudasirmirza** [comment]
My apologies, I posted this at a wrong place, I will close this.
